### PR TITLE
Events, part one

### PR DESCRIPTION
**Pull request by markspolakovs**

### CODE REVIEW

File: src/Classes/MyRadio/CoreUtils.php [hunk 1]
```diff
@@ -177,6 +177,23 @@ public static function getRfc2822Timestamp($time = null)
         return gmdate('r', $time);
     }
 
+    /**
+     * Returns an ISO 8601-formatted timestamp (for JavaScript).
+     *
+     * @param int $time The time to get the timestamp for. Default right now.
+     *
+     * @return string a timestamp
+     * @assert (30) == 'Thu, 01 Jan 1970 00:00:30 +0000'
```
Comment (markspolakovs) [inline src/Classes/MyRadio/CoreUtils.php:186]: Incorrect

**Comment by URY-GitHub** [comment]
Thank you for this pull request, @markspolakovs!

@michael-grace, would you mind reviewing this PR, please? Feel free to ask anyone for help if you're not quite sure what you're doing.

**Comment by markspolakovs** [comment]
Still a long way to go, but good enough for initial rollout (modulo a bit of CSS tweaking). Will almost certainly need a squash-merge though.

**Comment by markspolakovs** [comment]
Not too fond of the use of the term "master" in new code... might change that

File: src/Templates/Events/viewEvent.twig [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{% extends 'stripe.twig' %}
+
+{% block stripecontent %}
+    <h1>{{ event.Title() }}</h1>
+    <h2>{{ event.StartTime() | date('d F y H:i') }}&mdash;{{ event.EndTime() | date('d F y H:i') }}</h2>
```
Comment (markspolakovs) [inline src/Templates/Events/viewEvent.twig:5]: Might be nice to hide the date if the end is on the same day

File: src/Classes/ServiceAPI/MyRadio_Event.php [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        return self::resultSetToObjArray($rows);
+    }
+
+    public static function create($data = [])
+    {
+        // Validate
+        $requiredFields = ['title', 'description_html', 'start_time', 'end_time'];
+        foreach ($requiredFields as $field) {
+            if (!(isset($data[$field]))) {
+                throw new MyRadioException("Missing $field", 400);
```
Comment (michael-grace) [inline src/Classes/ServiceAPI/MyRadio_Event.php:175]: rip my missing $field. $ in string

Comment (markspolakovs) [reply src/Classes/ServiceAPI/MyRadio_Event.php:175]: it's a double-quote string, so this is fine :) it'll get replaced with e.g. `missing title`

File: src/Templates/Events/calendar.twig [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        document.addEventListener('DOMContentLoaded', function() {
+            var calendarEl = document.getElementById('calendar');
+            window.myradio_calendar = new FullCalendar.Calendar(calendarEl, {
+                eventSources: [
+                    {
+                        url: '{{ baseurl }}Events/a-getevents',
+                        startParam: "start",
+                        endParam: "end",
+                        eventDataTransform: function(obj) {
+                            console.log("transforming", obj);
```
Comment (mstratford) [inline src/Templates/Events/calendar.twig:26]: Just a tidy up here...

**Review by mstratford — APPROVED** [review]
LGTM

**Pull request merged**